src/Microsoft.Health.Fhir.SqlServer/Features/ChangeFeed/SqlServerFhirResourceChangeDataStore.cs

```diff
@@ -0,0 +1,134 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Data;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Abstractions.Data;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.SqlServer;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.ChangeFeed
+{
+    /// <summary>
+    /// A data store that provides functionalities to get resource changes for change feed sources.
+    /// </summary>
+    public class SqlServerFhirResourceChangeDataStore : IChangeFeedSource<ResourceChangeData>
+    {
+        private readonly ISqlConnectionFactory _sqlConnectionFactory;
+        private readonly ILogger<SqlServerFhirResourceChangeDataStore> _logger;
+        private ConcurrentDictionary<short, string> _resourceTypeIdToTypeName;
+
+        // dbnetlib error value for timeout expired
+        private const short TIMEOUTEXPIRED = -2;
```

Comment:
I realize this isn't a custom error code, but does it make sense to add this to `SqlErrorCodes.cs`?

Reply:
This change was merged into the healthcare shared components repo in [this PR](https://github.com/microsoft/healthcare-shared-components/pull/269). As discussed, since it isn't crucial that we pull this in now, we can just create a user story so we don't forget to update it later.

Reply:
Created a [user store](https://microsofthealth.visualstudio.com/Health/_workitems/edit/83513).